driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -83,7 +86,7 @@ public void apply(final AsyncAggregateResponseBatchCursor<RawBsonDocument> curso
 
     @Override
     public void close() {
-        if (!isClosed()) {
+        if (!isClosed.getAndSet(true) && noOperationInProgress.get()) {
```

Comment:
This is racy: What if right after getting a false value from `noOperationInProgress`, an operation is started?  In that case, we'll close the wrapped cursor and release the binding.  

We need to synchronize between here and starting an operation to ensure that can't happen.

Reply:
I now synchronize on the `isClosed`, `noOperationInProgress` and the new `closePending` variables.